### DISCUSSION TITLE
Smart Meter Interface: Script for ORNO OR-WE-517

### DISCUSSION THREAD

**Discussion by SigiSom** [arendst/Tasmota#18728] [Show and tell]
Hello, this is my working script for the ORNO OR-WE-517:

```
>D
>B
->sensor53 r
->sensor53 d0
>M 1
+1,3,M,0,9600,WE517,1(4),5,01030100,0103001C,0103000E,01030010,01030012,01030016,01030018,0103001A
1,010304ffffffff@i0:1,Energy Tot,kWh,Energy_Total,17
1,010304ffffffff@i1:1,Power,kW,Power_Total,19
1,010304ffffffff@i2:1,Voltage L1,V,Voltage_L1,17
1,010304ffffffff@i3:1,Voltage L2,V,Voltage_L2,17
1,010304ffffffff@i4:1,Voltage L3,V,Voltage_L3,17
1,010304ffffffff@i5:1,Current L1,A,Current_L1,18
1,010304ffffffff@i6:1,Current L2,A,Current_L2,18
1,010304ffffffff@i7:1,Current L3,A,Current_L3,18
#
```
I use a self designed PCB with a ESP07 module. TX is connected to GPIO1, RX to GPIO3 and GPIO4 is used to enable the RS485-Transmitter.

**Comment by anticharm01** [comment]
Hi,

Using a WiFi Adapter, but values are empty.

Dump mode shows values .

Thanks.

20:38:43.250 MQT: tele/tasmota_A31857/SENSOR = {"Time":"2024-08-31T20:38:43","WE517":{"Energy_Total":0.0,"Power_Total":0.000,"Voltage_L1":0.0,"Voltage_L2":0.0,"Voltage_L3":0.0,"Current_L1":0.00,"Current_L2":0.00,"Current_L3":0.00}}
20:38:43.407 > 01 03 00 10 00 02 c5 ce 
20:38:43.861 > 01 03 00 12 00 02 64 0e 
20:38:44.361 > 01 03 00 16 00 02 25 cf 
20:38:44.867 > 01 03 00 18 00 02 44 0c 
20:38:45.372 > 01 03 00 1a 00 02 e5 cc 
20:38:45.876 > 01 03 01 00 00 02 c5 f7 
20:38:46.382 > 01 03 00 1c 00 02 05 cd 
20:38:46.887 > 01 03 00 0e 00 02 a5 c8 
20:38:47.392 > 01 03 00 10 00 02 c5 ce 
20:38:47.897 > 01 03 00 12 00 02 64 0e 
20:38:48.403 > 01 03 00 16 00 02 25 cf 
20:38:48.909 > 01 03 00 18 00 02 44 0c 
20:38:49.362 > 01 03 00 1a 00 02 e5 cc 
20:38:49.867 > 01 03 01 00 00 02 c5 f7 
20:38:50.372 > 01 03 00 1c 00 02 05 cd 
20:38:50.876 > 01 03 00 0e 00 02 a5 c8 
20:38:51.383 > 01 03 00 10 00 02 c5 ce 
20:38:51.887 > 01 03 00 12 00 02 64 0e 
20:38:52.381 > 01 03 00 16 00 02 25 cf 
20:38:52.888 > 01 03 00 18 00 02 44 0c 
20:38:53.402 > 01 03 00 1a 00 02 e5 cc 
20:38:53.906 > 01 03 01 00 00 02 c5 f7 
20:38:54.362 > 01 03 00 1c 00 02 05 cd 
20:38:54.867 > 01 03 00 0e 00 02 a5 c8 
20:38:55.361 > 01 03 00 10 00 02 c5 ce 
20:38:55.865 > 01 03 00 12 00 02 64 0e 
20:38:56.372 > 01 03 00 16 00 02 25 cf 
20:38:56.879 > 01 03 00 18 00 02 44 0c 
20:38:57.380 > 01 03 00 1a 00 02 e5 cc 
20:38:57.884 > 01 03 01 00 00 02 c5 f7 
20:38:58.393 > 01 03 00 1c 00 02 05 cd 
20:38:58.901 > 01 03 00 0e 00 02 a5 c8 
20:38:59.362 > 01 03 00 10 00 02 c5 ce 
20:38:59.867 > 01 03 00 12 00 02 64 0e 
20:39:00.370 > 01 03 00 16 00 02 25 cf 
20:39:00.877 > 01 03 00 18 00 02 44 0c 
20:39:01.371 > 01 03 00 1a 00 02 e5 cc 
20:39:01.877 > 01 03 01 00 00 02 c5 f7 
20:39:02.383 > 01 03 00 1c 00 02 05 cd 
20:39:02.891 > 01 03 00 0e 00 02 a5 c8 
20:39:03.404 > 01 03 00 10 00 02 c5 ce 
20:39:03.910 > 01 03 00 12 00 02 64 0e 
20:39:04.362 > 01 03 00 16 00 02 25 cf 
20:39:04.868 > 01 03 00 18 00 02 44 0c 
20:39:05.373 > 01 03 00 1a 00 02 e5 cc 
20:39:05.481 CMD: sensor53 d0

**Reply by gemu2015** [reply]
these dump values marked with `>` are requests send to your meter and there is NO response from the meter.
check LEDs on your RS485 adapter RX and TX LEDs must blink on successful communication.

**Reply by anticharm01** [reply]
Hmm, the clear led flashes.
![VideoCapture_20240901-085239](https://github.com/user-attachments/assets/96f2f09d-5c63-4aa8-921a-dc93e05b05f5)

**Reply by gemu2015** [reply]
The above script is for a MODBUS meter. This optical head does not seem to be ment for MODBUS ?

**Comment by anticharm01** [comment]
Ahhh, uhhhm.

Docu says "(SML, OBIS-) signal"...

Good point.

**Comment by anticharm01** [comment]
This is beyond my capabilities. I have no idea how to get a single bit out of that meter via infrared.

The manual says

"It corresponds to the communication standard IR (near infrared) and RS485. IR meets the requirements of the protocol EN62056(IEC1107), and the communication RS485 uses the protocol MODBUS."

What is "EN62056", is it OBIS?

How do i get this meter to send something or how to get to know it is sending?

thanks.

**Comment by gemu2015** [comment]
for MODBUS which i would prefer you need an RS485 to TTL Adapter, then connect the meter via wires and use the above script

https://de.aliexpress.com/item/1005006941264328.html?src=google&src=google&albch=shopping&acnt=272-267-0231&isdl=y&slnk=&plac=&mtctp=&albbt=Google_7_shopping&aff_platform=google&aff_short_key=UneMJZVf&gclsrc=aw.ds&&albagn=888888&&ds_e_adid=&ds_e_matchtype=&ds_e_device=c&ds_e_network=x&ds_e_product_group_id=&ds_e_product_id=de1005006941264328&ds_e_product_merchant_id=5332406508&ds_e_product_country=DE&ds_e_product_language=de&ds_e_product_channel=online&ds_e_product_store_id=&ds_url_v=2&albcp=20542169885&albag=&isSmbAutoCall=false&needSmbHouyi=false&gad_source=1&gbraid=0AAAAAoukdWPj8oRa5Uh9dZvcT8ik_1222&gclid=EAIaIQobChMI2JqQqfOhiAMVqfF5BB0ICgY4EAQYBiABEgKI8_D_BwE

**Comment by anticharm01** [comment]
yeah, sigh.

Thought it was such a simple thing. The meter has infrared, the small reader has an ESP with WIFI on board.

     flash Tasmota - add to Home assistant - everybody happy. FAILED.

Maybe I try to drop the led header and use the ESP to read directly from ModBus. 

Thanks

**Reply by anticharm01** [reply]
Another thing, that came in my mind... maybe infrared has to be enabled somehow. Manual says nothing. There is some software, somewhere... and I need... a cable. Damn.

**Reply by gemu2015** [reply]
if you get to know the exact protocol of the IR interface you could also use this with your IR head, but i could not find any further info about this meter.

**Reply by anticharm01** [reply]
I'm trying to find the original software. It's not on the homepage but I saw a link in one of the forums.

Have another approach with a Hi-Flying Protoss PW11. Ordered it minutes ago. :)

Thanks

**Comment by anticharm01** [comment]
Found something :)
[three phase four modular PC SOFTWRAE manual.docx](https://github.com/user-attachments/files/16836654/three.phase.four.modular.PC.SOFTWRAE.manual.docx)

**Comment by anticharm01** [comment]
Could not connect via IR. Tried IR to USB head, too. 

Finally ended with a PW11-H ModBus to TCP. 

Working fine.
